sdk/src/test/java/io/radar/sdk/matchers/RangeMatcher.kt

```diff
@@ -0,0 +1,101 @@
+package io.radar.sdk.matchers
+
+import org.hamcrest.BaseMatcher
+import org.hamcrest.Description
+import java.lang.UnsupportedOperationException
+import java.util.*
+
+/**
+ * Used to provide matchers for some range of object
+ */
+sealed class RangeMatcher<T>(val from: T, val to: T, val inclusive: Boolean) : BaseMatcher<T>() {
+
+    companion object {
+        fun isBetween(from: Date, to: Date, inclusive: Boolean = true): RangeMatcher<Date> {
+            return DateRangeMatcher(from, to, inclusive)
+        }
+        fun isBetween(from: Number, to: Number, inclusive: Boolean = true): RangeMatcher<Number> {
+            return NumberMatcher(from, to, inclusive)
+        }
+        fun isGreaterThan(from: Number): RangeMatcher<Number> {
+            return NumberMatcher(from)
+        }
+        fun isLessThan(to: Number): RangeMatcher<Number> {
+            return NumberMatcher(to = to)
+        }
+    }
+
+    private class NumberMatcher(
+        from: Number = Double.MIN_VALUE,
+        to: Number = Double.MAX_VALUE,
+        inclusive: Boolean = false
+    ) : RangeMatcher<Number>(from, to, inclusive) {
+        override fun matches(item: Any?): Boolean {
+            return when (item) {
+                null -> {
+                    false
+                }
+                is Number -> {
+                    if (inclusive) {
+                        item >= from && item <= to
+                    } else {
+                        item > from && item < to
+                    }
+                }
+                else -> {
+                    false
+                }
+            }
+        }
+    }
+
+    /**
+     * Matcher for checking if a given [Date] is within the bounds of other dates
+     */
+    private class DateRangeMatcher(
+        from: Date,
+        to: Date,
+        inclusive: Boolean
+    ) : RangeMatcher<Date>(from, to, inclusive) {
+        override fun matches(item: Any?): Boolean {
+            return when (item) {
+                null -> {
+                    false
+                }
+                is Date -> {
+                    if (inclusive) {
+                        item in from..to
+                    } else {
+                        item > from && item < to
+                    }
+                }
```

Comment:
Would it be easier just to convert the `Date` to a `Long` and use the comparison for `Long`s (or vice versa)?

Reply:
I don't think it would necessarily be easier. it is another option for sure.